lib/ja_serializer/dsl.ex

```diff
@@ -250,7 +255,7 @@ defmodule JaSerializer.DSL do
         @compile {:inline, [{att, 1}, {att, 2}]}
 
         def unquote(att)(m), do: Map.get(m, unquote(att))
-        def unquote(att)(m, c), do: unquote(att)(m)
+        def unquote(att)(m, c), do: __MODULE__.unquote(att)(m)
```

Comment:
@marpo60 I figured out what the issue is. In my app, we have the view functions as private functions. So by changing this to `__MODULE__.unquote`, the private functions can no longer be called. I'm not sure if it's possible, but I would love to retain support for defining the attribute functions as private. 

I pushed up a test that reproduces the issue: https://github.com/vt-elixir/ja_serializer/compare/private-attribute-fn

Let me know if you're interested in fixing, otherwise I'll take a stab at it later this week.

Reply:
I dug in a little more and it looks like we only supported the single arity version as private, but the two arity functions had to be public. This was not documented and not officially supported, so I think we're good with your change. The single version just happened to support being private because of how it was being called, but this was inconsistent.

I'll add a note in the changelog about this being a potentially breaking change and that users just need to make their private functions public.

Thanks again!

Reply:
@beerlington I investigate a bit yesterday night and here are my findings.

I added this test:

```
  test "x" do
    defmodule PrivateArticle do
      use JaSerializer

      attributes([:private_arity_one, :private_with_two])

      defp private_arity_one(_a), do: "private"
      defp private_arity_two(_a, _b), do: "private"
    end

    article = %TestModel.Article{}
    assert PrivateArticle.attributes(article, %{}) == %{
      private_arity_one: "private",
      private_arity_two: "private"
    }
  end
```

I tested this test in this commit https://github.com/vt-elixir/ja_serializer/commit/b5d9f1d right before https://github.com/vt-elixir/ja_serializer/pull/245 got merged.

The test is not working.

```
  1) test x (JaSerializer.SerializerTest)
     test/ja_serializer/serializer_test.exs:41
     ** (UndefinedFunctionError) function JaSerializer.SerializerTest.PrivateArticle.private_arity_one/1 is undefined or private
```

I then tested it after that PR got merged https://github.com/vt-elixir/ja_serializer/commit/11e79da

The test is green.

IMHO this was just a side effect, and a not intended one caused by changing this:
https://github.com/vt-elixir/ja_serializer/pull/245/files#diff-bbdfd0bc4622bc60b9c795038912ad87L228

Having the definition of attributes as private function will conflict with inline by compiler methods, so I think it should be a must that if you overwrite an attribute it should with a public function.

Cheers.